<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dock_Script_Log</class>
 <widget class="QDockWidget" name="Dock_Script_Log">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>289</width>
    <height>233</height>
   </rect>
  </property>
  <property name="windowIcon">
   <iconset theme="utilities-terminal">
    <normaloff/>
   </iconset>
  </property>
  <property name="windowTitle">
   <string>Script Console</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QFrame" name="frame_advanced">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QFrame" name="frame_editor_buttons">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QToolButton" name="button_new">
             <property name="text">
              <string>&amp;New</string>
             </property>
             <property name="icon">
              <iconset theme="document-new"/>
             </property>
             <property name="toolButtonStyle">
              <enum>Qt::ToolButtonIconOnly</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="button_open">
             <property name="text">
              <string>&amp;Open</string>
             </property>
             <property name="icon">
              <iconset theme="document-open"/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="button_save">
             <property name="text">
              <string>&amp;Save</string>
             </property>
             <property name="icon">
              <iconset theme="document-save"/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="Line" name="line">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="button_undo">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>&amp;Undo</string>
             </property>
             <property name="icon">
              <iconset theme="edit-undo"/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="button_redo">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>R&amp;edo</string>
             </property>
             <property name="icon">
              <iconset theme="edit-redo"/>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QToolButton" name="button_run">
             <property name="text">
              <string>&amp;Run</string>
             </property>
             <property name="icon">
              <iconset theme="debug-run"/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="Script_Edit" name="source_editor">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>24</height>
             </size>
            </property>
            <property name="lineWrapMode">
             <enum>QPlainTextEdit::NoWrap</enum>
            </property>
            <property name="tabStopWidth">
             <number>20</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="button_run_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Run</string>
            </property>
            <property name="icon">
             <iconset theme="debug-run"/>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_basic">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="button_clear_output">
            <property name="text">
             <string>Clear</string>
            </property>
            <property name="icon">
             <iconset theme="edit-clear-list">
              <normaloff/>
             </iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="button_stop">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="toolTip">
             <string>Stop the currently running script</string>
            </property>
            <property name="text">
             <string>Abort</string>
            </property>
            <property name="icon">
             <iconset theme="process-stop">
              <normaloff/>
             </iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Copyable_Text_Browser" name="text_output">
          <property name="palette">
           <palette>
            <active>
             <colorrole role="Text">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>255</green>
                <blue>255</blue>
               </color>
              </brush>
             </colorrole>
             <colorrole role="Base">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>0</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="Text">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>255</green>
                <blue>255</blue>
               </color>
              </brush>
             </colorrole>
             <colorrole role="Base">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>0</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="Text">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>190</red>
                <green>190</green>
                <blue>190</blue>
               </color>
              </brush>
             </colorrole>
             <colorrole role="Base">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>239</red>
                <green>235</green>
                <blue>231</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
          <property name="font">
           <font>
            <family>Monospace</family>
           </font>
          </property>
          <property name="undoRedoEnabled">
           <bool>false</bool>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::LinksAccessibleByKeyboard|Qt::LinksAccessibleByMouse|Qt::TextBrowserInteraction|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Copyable_Text_Browser</class>
   <extends>QTextBrowser</extends>
   <header>copyable_text_browser.hpp</header>
  </customwidget>
  <customwidget>
   <class>Script_Edit</class>
   <extends>QPlainTextEdit</extends>
   <header>script_edit.hpp</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>text_output</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>button_clear_output</sender>
   <signal>clicked()</signal>
   <receiver>text_output</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_run_2</sender>
   <signal>clicked()</signal>
   <receiver>button_run</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>262</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>263</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_undo</sender>
   <signal>clicked()</signal>
   <receiver>source_editor</receiver>
   <slot>undo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>40</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_redo</sender>
   <signal>clicked()</signal>
   <receiver>source_editor</receiver>
   <slot>redo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>37</y>
    </hint>
    <hint type="destinationlabel">
     <x>154</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>source_editor</sender>
   <signal>undoAvailable(bool)</signal>
   <receiver>button_undo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>source_editor</sender>
   <signal>redoAvailable(bool)</signal>
   <receiver>button_redo</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>191</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_new</sender>
   <signal>clicked()</signal>
   <receiver>source_editor</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>26</x>
     <y>37</y>
    </hint>
    <hint type="destinationlabel">
     <x>43</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
